src/biokbase/narrative/tests/test_app_util.py

```diff
@@ -2,182 +2,239 @@
-                        "input_parameter": "workspace",
-                    },
-                    {
-                        "target_position": 1,
-                        "target_property": "foo",
-                        "input_parameter": "baz",
-                    },
-                    {
-                        "target_position": 2,
-                        "input_parameter": "ref_input",
-                        "target_type_transform": "ref",
-                    },
-                    {
-                        "target_position": 3,
-                        "input_parameter": "a_list",
-                        "target_type_transform": "list<ref>",
-                    },
-                    {
-                        "target_position": 4,
-                        "input_parameter": "a_num",
-                        "target_type_transform": "int",
-                    },
-                ],
-            }
-        }
-        expected = {
-            "an_input": "input1",
-            "workspace": "my_workspace",
-            "baz": "bar",
-            "ref_input": "obj_id",
-            "a_list": ["num_1", "num_2", "num_3"],
-            "a_num": 123,
-        }
-        self.assertDictEqual(map_inputs_from_job(inputs, app_spec), expected)
-
-    def test_map_outputs_from_state_simple(self):
-        os.environ["KB_WORKSPACE_ID"] = self.workspace
-        app_spec = {
-            "parameters": [],
-            "behavior": {
-                "output_mapping": [{"narrative_system_variable": "workspace"}]
-            },
-        }
-        self.assertTupleEqual(
-            map_outputs_from_state(None, None, app_spec),
-            ("kbaseDefaultNarrativeOutput", self.workspace),
-        )
-
-    def test_map_outputs_from_state(self):
-        os.environ["KB_WORKSPACE_ID"] = self.workspace
-        app_spec = {
-            "widgets": {"input": None, "output": "testOutputWidget"},
-            "parameters": [],
-            "behavior": {
-                "kb_service_output_mapping": [
-                    {"narrative_system_variable": "workspace", "target_property": "ws"},
-                    {"constant_value": 5, "target_property": "a_constant"},
-                    {
-                        "service_method_output_path": [1],
-                        "target_property": "a_path_ref",
-                    },
-                    {"input_parameter": "an_input", "target_property": "an_input"},
-                ]
-            },
+from .narrative_mock.mockclients import get_mock_client
+
+user_token = "SOME_TOKEN"
+config = util.ConfigTests()
+user_id = config.get("users", "test_user")
+user_token = util.read_token_file(config.get_path("token_files", "test_user", from_root=True))
+
+good_tag = "release"
+bad_tag = "not_a_tag"
+# inject phony variables into the environment
+bad_fake_token = "NotAGoodTokenLOL"
+workspace = "valid_workspace"
+bad_variable = "not a variable"
+
+def test_check_tag_good():
```

Comment:
Most of the changes here are reflected by updating to using pytest and asserts instead of unittest's class-based stuff.